server/app/models/distributed_lock.rb

```diff
@@ -13,17 +15,16 @@ def self.with_lock(name, timeout = 60)
     lock_id = nil
     begin
       if timeout.to_f > 0.0
-        Timeout.timeout(timeout) do
-          sleep 0.01 until lock_id = self.obtain_lock(name)
-        end
+        lock_id = WaitHelper.wait(timeout: timeout) { self.obtain_lock(name) }
```

Comment:
This changes the default `sleep` interval from 0.01 to 0.5. Looking at the calls to `with_dlock` in the server:

```ruby
    with_dlock("check_deploy_queue:#{service_deploy.grid_service_id}", 10) do
    with_dlock(DLOCK_KEY, nil) do
    with_dlock('mongodb_migrate', 60) do
```

I don't suppose this makes tooooooo much of a difference?